common/src/main/java/com/microsoft/identity/common/internal/platform/AndroidPlatformUtil.java

```diff
@@ -261,6 +263,31 @@ public static ArrayList<Map.Entry<String, String>> updateWithOrDeleteWebAuthnPar
         return result;
     }
 
+    /**
+     * Check if the host app is running within work profile.
+     * @param appContext current application context.
+     * @return true if app is in work profile, false if in personal profile or OS is below LOLLIPOP.
+     */
+    public static boolean isInWorkProfile(@NonNull final Context appContext) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            final UserManager um = (UserManager) appContext.getSystemService(Context.USER_SERVICE);
+            return um.isManagedProfile();
```

Comment:
> return um.isManagedProfile(); [](http://example.com/codeflow?start=11&length=30)

if the device is fully managed, would this return false?

Reply:
Talked to Mohit about this offline; the thinking is that for COBO devices, since they would constitute as a managed device (and not a managed profile), this method would return false. One Intune engineer also believes this is the case, but I will also check in with a different Intune team.
This question also prompted a renaming from isInWorkProfile to isInManagedProfile, to avoid any confusion as to what constitutes as a work profile.

Reply:
(Link Mohit found: https://developers.google.com/android/work/play/emm-api/management-types#devices_and_management_types)